hr_attendance_management/models/hr_holidays.py

```diff
@@ -10,15 +10,29 @@
 ##############################################################################
 import logging
 
+from datetime import datetime
 from odoo import models, fields, api
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+import pytz
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as DATETIME_FORMAT
 
 _logger = logging.getLogger(__name__)
 
 
 class HrHolidays(models.Model):
     _inherit = 'hr.holidays'
 
-    keep_due_hours = fields.Boolean(related='holiday_status_id.keep_due_hours')
+    @api.multi
+    @api.depends('number_of_days_temp', 'type')
+    def _compute_number_of_days(self):
+        for holiday in self:
+            # Check that it isn't possible to take less than half a day.
+            if holiday.number_of_days_temp % 0.5 != 0:
+                raise ValidationError(_(
```

Comment:
The best place to do this verification would be to inherit this method instead (found in hr_holidays module):
```
@api.constrains('state', 'number_of_days_temp', 'holiday_status_id')
    def _check_holidays(self):
```